unstructured/ingest/connector/sql/connector.py

```diff
@@ -0,0 +1,200 @@
+                data["coordinates_id"] = coordinates_id
+
+        # Dict as string formatting
+        if record_locator := data.get("metadata", {}).get("data_source", {}).get("record_locator"):
+            # Explicit casting otherwise fails schema type checking
+            data["metadata"]["data_source"]["record_locator"] = str(json.dumps(record_locator))
+
+        # Array of items as string formatting
+        if points := data.get("metadata", {}).get("coordinates", {}).get("points"):
+            data["metadata"]["coordinates"]["points"] = str(json.dumps(points))
+
+        if links := data.get("metadata", {}).get("links", {}):
+            data["metadata"]["links"] = str(json.dumps(links))
+
+        if permissions_data := (
+            data.get("metadata", {}).get("data_source", {}).get("permissions_data")
+        ):
+            data["metadata"]["data_source"]["permissions_data"] = json.dumps(permissions_data)
+
+        # Datetime formatting
+        if date_created := data.get("metadata", {}).get("data_source", {}).get("date_created"):
+            data["metadata"]["data_source"]["date_created"] = datetime.fromisoformat(date_created)
+
+        if date_modified := data.get("metadata", {}).get("data_source", {}).get("date_modified"):
+            data["metadata"]["data_source"]["date_modified"] = datetime.fromisoformat(date_modified)
+
+        if date_processed := data.get("metadata", {}).get("data_source", {}).get("date_processed"):
+            data["metadata"]["data_source"]["date_processed"] = datetime.fromisoformat(
+                date_processed
+            )
+
+        if last_modified := data.get("metadata", {}).get("last_modified", {}):
+            data["metadata"]["last_modified"] = datetime.fromisoformat(last_modified)
+
+        # String casting
+        if version := data.get("metadata", {}).get("data_source", {}).get("version"):
+            data["metadata"]["data_source"]["version"] = str(version)
+
+        if page_number := data.get("metadata", {}).get("page_number"):
+            data["metadata"]["page_number"] = str(page_number)
+
+        if regex_metadata := data.get("metadata", {}).get("regex_metadata"):
+            data["metadata"]["regex_metadata"] = str(json.dumps(regex_metadata))
+
+        data_source = data.get("metadata", {}).pop("data_source", None)
+        coordinates = data.get("metadata", {}).pop("coordinates", None)
+        metadata = data.pop("metadata", None)
+
+        return data, metadata, data_source, coordinates
+
+    # @DestinationConnectionError.wrap
+    def write_dict(self, *args, json_list: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
+        from sqlalchemy import insert
+
+        logger.info(
+            f"writing {len(json_list)} objects to database url " f"{self.connector_config.db_url} "
+        )
+
+        elements = make_elements_table(
+            self.write_config.table_name_mapping.get(ELEMENTS_TABLE_NAME, ELEMENTS_TABLE_NAME)
```

Comment:
Is there a specific reason were using .get like this? (if it doesn't find ELEMENTS_TABLE_NAME, it use ELEMENTS_TABLE_NAME). Probably be better to just use ["ELEMENTS_TABLE_NAME"] and  hard fail if there is no ELEMENTS_TABLE_NAME.